src/main/java/io/pravega/perf/PravegaStreamHandler.java

```diff
@@ -135,15 +141,25 @@ void recreate() throws InterruptedException, ExecutionException, TimeoutExceptio
     }
 
     ReaderGroup createReaderGroup(boolean reset) throws URISyntaxException {
-        final ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope,
-                ClientConfig.builder().controllerURI(new URI(controllerUri)).build());
-        final ReaderGroupConfig rdGrpConfig = ReaderGroupConfig.builder()
-                            .stream(Stream.of(scope, stream)).build();
-        readerGroupManager.createReaderGroup(stream, rdGrpConfig);
-        final ReaderGroup rdGroup = readerGroupManager.getReaderGroup(stream);
+        if (readerGroupManager == null) {
+            readerGroupManager = ReaderGroupManager.withScope(scope,
+                    ClientConfig.builder().controllerURI(new URI(controllerUri)).build());
+            rdGrpConfig = ReaderGroupConfig.builder()
+                    .stream(Stream.of(scope, stream)).build();
+        }
+        readerGroupManager.createReaderGroup(rdGrpName, rdGrpConfig);
+        final ReaderGroup rdGroup = readerGroupManager.getReaderGroup(rdGrpName);
         if (reset) {
             rdGroup.resetReaderGroup(rdGrpConfig);
         }
         return rdGroup;
     }
+
+    void deleteReaderGroup() {
+        try {
+            readerGroupManager.deleteReaderGroup(rdGrpName);
+        } catch (InvalidStreamException e) {
+            System.out.println("Reader Group is already deleted");
```

Comment:
In the absence of any context this message would be confusing. Perhaps: "Cannot delete reader group X because it is already deleted" would be better.

Reply:
yes.. its changed now..